flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/JobManager.java

```diff
@@ -374,6 +375,8 @@ public JobSubmissionResult submitJob(JobGraph job) throws IOException {
 				LOG.debug(String.format("Running master initialization of job %s (%s)", job.getJobID(), job.getName()));
 			}
 
+			final int numSlots = scheduler.getTotalNumberOfSlots();
```

Comment:
Shouldn't this be set to `getNumberOfAvailableSlots()` for the PARALLELISM_AUTO_MAX case?